pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -208,7 +205,9 @@ public void onBecomeDroppedFromOffline(Message message, NotificationContext cont
       try {
         segmentLock.lock();
 
-        final File segmentDir = new File(_fetcherAndLoader.getSegmentLocalDirectory(tableNameWithType, segmentName));
+        TableDataManager tableDataManager = _instanceDataManager.getTableDataManager(tableNameWithType);
+        Preconditions.checkNotNull(tableDataManager);
```

Comment:
TableDataManager might not exist if there is no segment loaded for this table yet. Can we read the segment data dir from the `_instanceDataManager`?